File: lesson_18/src/main/java/org/lesson18/Customer.java
package org.lesson18;


import jakarta.persistence.*;

import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    private UUID id;
    @Column(name = "fio")
    private String fio;

    @ManyToMany
    @JoinTable(
            name = "purchase_details",
            joinColumns = @JoinColumn(name = "customer_id"),
            inverseJoinColumns = @JoinColumn(name = "product_id")
    )
    private List<Product> products;

    @OneToMany

    public List<Product> getProducts() {
        return products;
    }

    public UUID getId() {
        return id;
    }

    public String getFio() {
        return fio;
    }

    public void setFio(String fio) {
        this.fio = fio;
    }

    public Customer(String fio) {
        super();
        this.fio = fio;
    }

    public Customer() {
    }
}

File: lesson_18/src/main/kotlin/org/lesson18kt/Customer.kt
package org.lesson18kt

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.JoinTable
import jakarta.persistence.ManyToMany
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "customer")
data class Customer(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    var id: UUID? = null,
    @Column(name = "fio")
    var fio: String? = null,

    @ManyToMany
    @JoinTable(
        name = "purchase_details",
        joinColumns = [JoinColumn(name = "customer_id")],
        inverseJoinColumns = [JoinColumn(name = "product_id")]
    )
    var products: List<Product>? = null
)
